source/agora/consensus/protocol/Nominator.d

```diff
@@ -1014,12 +1012,14 @@ extern(D):
                     assert(0, format!"combineCandidates: Invalid consensus data: %s"(
                         msg));
 
-                log.info("combineCandidates: {}", slot_idx);
                 log.trace("Combined consensus data: {}", data.prettify);
-                // todo: currently we just pick the first of the candidate values,
-                // but we should ideally pick tx's out of the combined set
-                return duplicate_value(&candidate);
+                values ~= data;
             }
+            // Nomination MUST be deterministic so we take the least number of validators missing pre-images
+            auto combined = values.sort!("a.missing_validators.length < b.missing_validators.length").front();
```

Comment:
do we want to handle the ties when the length of the missing validators is the same, but the contents is not? we could just do a < b on the sorted values of 'a' and 'b' as < is defined on the arrays